### PR TITLE
have commands that can be run by remote daemon

### PR DESCRIPTION
**Pull request by josecv**
## Description

A few sentences describing the overall goals of the pull request's commits.

## Checklist

 - [ ] I made sure to update `./CHANGELOG.md`.
 - [ ] I made sure to add any docs changes required for my change (including release notes).
 - [x] My change is adequately tested.
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.
 - [ ] I updated `TELEMETRY.md` if I added, changed, or removed a metric name.
 - [ ] Once my PR is ready to have integration tests ran, I posted the PR in #telepresence-dev in the datawire-oss slack so that the "ok to test" label can be applied.

### CODE REVIEW

**Review by thallgren — COMMENTED** [review]
This is great! Aside from a more static way of handling stdout/stderr (not that important), it's precisely what I envisioned.

**Comment by josecv** [comment]
I'm not adding a changelog entry cause I guess this isn't really visible to users yet, but let me know if you think I should add one

File: pkg/client/cli/cliutil/flagtypes.go [hunk 1]
```diff
@@ -121,9 +139,9 @@ var flagTypes = map[TypeEnum]interface{}{
 	"int32": func(fs *pfs, x *int32) Value { fs.Int32Var(x, "x", *x, ""); return simple(fs) },
 	"int64": func(fs *pfs, x *int64) Value { fs.Int64Var(x, "x", *x, ""); return simple(fs) },
 
-	"int-slice":   func(fs *pfs, x *[]int) Value { fs.IntSliceVar(x, "x", *x, ""); return slice(fs, x) },
-	"int32-slice": func(fs *pfs, x *[]int32) Value { fs.Int32SliceVar(x, "x", *x, ""); return slice(fs, x) },
-	"int64-slice": func(fs *pfs, x *[]int64) Value { fs.Int64SliceVar(x, "x", *x, ""); return slice(fs, x) },
+	"intSlice":   func(fs *pfs, x *[]int) Value { fs.IntSliceVar(x, "x", *x, ""); return slice(fs, x) },
```
Comment (josecv) [inline pkg/client/cli/cliutil/flagtypes.go:142]: these camelCasing changes were necessary to line up with the type reported by pflag

File: pkg/client/cli/command_group.go [hunk 2]
```diff
@@ -64,15 +53,15 @@ func init() {
...
 }
 
-// AddCommandGroups adds all the groups in the given CommandGroup to the command,  replaces
+// AddCommandGroups adds all the groups in the given CommandGroups to the command,  replaces
 // the its standard usage template with a template that groups the commands according to that group.
-func AddCommandGroups(cmd *cobra.Command, groups []CommandGroup) {
+func AddCommandGroups(cmd *cobra.Command, groups cliutil.CommandGroups) {
 	for _, group := range groups {
-		cmd.AddCommand(group.Commands...)
+		cmd.AddCommand(group...)
```
Comment (maxlegault) [inline pkg/client/cli/command_group.go]: For clarity, could we rename `group` to `commands`?

File: pkg/client/cli/command_group.go [hunk 3]
```diff
@@ -90,8 +79,8 @@ Examples:
 
 Available Commands:
 {{- if commandGroups .}}
-{{- range $group := commandGroups .}}
-  {{$group.Name}}:{{range $group.Commands}}
+{{- range $name, $group := commandGroups .}}
```
Comment (maxlegault) [inline pkg/client/cli/command_group.go]: Same comment as above here, it think it would be clearer to rename `$group` to `$commands`

**Review by maxlegault — APPROVED** [review]
Two nit-picks, but it all LGTM!

**Pull request merged**